shared/utility/motion/ForwardKinematics.h

```diff
@@ -324,61 +318,198 @@ namespace motion {
+                com << particle.x(), particle.y(), particle.z(), 1.0;
+                double mass = particle.w();
+
+                // Calculate CoM in torso space
+                com                       = Htp * com;
+                Eigen::Vector4d com_world = Hwt * com;
+
+                return std::pair<Eigen::Vector3d, double>{Eigen::Vector3d(com.x(), com.y(), com.z()), mass};
+            };
+
+            // Get the centre of mass for each particle in torso space
+            // There are 16 particles in total
+            std::array<std::pair<Eigen::Vector3d, double>, 16> particles = {
+                com("Head", forward_kinematics[utility::input::ServoID::HEAD_PITCH], model.massModel.head),
+                com("Left Upper Arm",
+                    forward_kinematics[utility::input::ServoID::L_SHOULDER_PITCH],
+                    model.massModel.arm_upper),
+                com("Right Upper Arm",
+                    forward_kinematics[utility::input::ServoID::R_SHOULDER_PITCH],
+                    model.massModel.arm_upper),
+                com("Left Lower Arm",
+                    forward_kinematics[utility::input::ServoID::L_SHOULDER_ROLL],
+                    model.massModel.arm_lower),
+                com("Right Lower Arm",
+                    forward_kinematics[utility::input::ServoID::R_SHOULDER_ROLL],
+                    model.massModel.arm_lower),
+                com("Left Hip", forward_kinematics[utility::input::ServoID::L_HIP_ROLL], model.massModel.hip_block),
+                com("Right Hip", forward_kinematics[utility::input::ServoID::R_HIP_ROLL], model.massModel.hip_block),
+                com("Left Upper Leg",
+                    forward_kinematics[utility::input::ServoID::L_HIP_PITCH],
+                    model.massModel.leg_upper),
+                com("Right Upper Leg",
+                    forward_kinematics[utility::input::ServoID::R_HIP_PITCH],
+                    model.massModel.leg_upper),
+                com("Left Lower Leg", forward_kinematics[utility::input::ServoID::L_KNEE], model.massModel.leg_lower),
+                com("Right Lower Leg", forward_kinematics[utility::input::ServoID::R_KNEE], model.massModel.leg_lower),
+                com("Left Ankle",
+                    forward_kinematics[utility::input::ServoID::L_ANKLE_PITCH],
+                    model.massModel.ankle_block),
+                com("Right Ankle",
+                    forward_kinematics[utility::input::ServoID::R_ANKLE_PITCH],
+                    model.massModel.ankle_block),
+                com("Left Foot", forward_kinematics[utility::input::ServoID::L_ANKLE_ROLL], model.massModel.foot),
+                com("Right Foot", forward_kinematics[utility::input::ServoID::R_ANKLE_ROLL], model.massModel.foot),
+                std::pair<Eigen::Vector3d, double>{Eigen::Vector3d::Zero(), 0.0},
+            };
+
+            // Add the torso into the list of particles, if requested
+            if (include_torso) {
```

Comment:
Just remove the include torso code, who knows what it was originally intended for